Formatting the code from data file.

ramon-bernardo opened the pull request:
### Pull Request Prelude
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Rename files.
- Change ticks time format.
- Add names next to item ids.
- Remove double/whitespaces.
- Single quotes to double quotes.
#### Move
- itemid > getId()
- uid > getUniqueId()
- type > getSubType()
- actionid > getActionId()
#### Improve
- Code logic, without else.
- Return value of functions.
- Boolean checks and comparisons.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

EPuncker commented:
that is nice! if you are accepting suggestions, I have 4:

1. add space between **if** and **(**. ex: data/talkactions/scripts/magiceffect.lua line 7
2. change file names to **snake_case** instead of **snakecase**
3. add new lines at end of the files that are missing it. ex: data/scripts/actions/others/costume_bag.lua
4. using `"` double quotes instead of single quotes `'` where applicable

ramon-bernardo commented:
@EPuncker **nice**!

Your suggestions are good;
1 - I removed the `()` from if's.
2 - Im going to do this now, but I have my doubts with some spacing. 
3 - This file already contains the space at the end, you say below the `for`? 
4 - I changed all single quotes, I kept only the sql. (Good mentioned it. I wouldn't remember)

My suggestions;
Change boolean checks and comparisons.
- `if bSomething == true` to `if bSomething` and `if bSomething == false` to `if not bSomething`
- `if myPtr ~= nil` to `if myPtr` and `if myPtr == nil` to `if not myPtr`

EPuncker commented:
> My suggestions;
> Change boolean checks and comparisons.
> 
> * `if bSomething == true` to `if bSomething` and `if bSomething == false` to `if not bSomething`
> * `if myPtr ~= nil` to `if myPtr` and `if myPtr == nil` to `if not myPtr`

I agree with this, but I'm going to ask @DSpeichert for his opinion as well

ramon-bernardo commented:
> I agree with this, but I'm going to ask @DSpeichert for his opinion as well

Okay, I'll be waiting DSpeichert opinion.
I kept the suggestions in separate commits.

DSpeichert commented:
Those are equivalent in Lua, so short form is fine.

ramon-bernardo commented:
I believe that all formatting is finished.
Some revisions are now needed.

Regarding suggestion 2 of @EPuncker
-  change file names to snake_case instead of snakecase

I believe that a new pull request is necessary. Why? renaming files makes the changes complete and not just lines. To review the files, it would be necessary to compare them one by one.

EPuncker commented:
forgot to remove the `\'` now that we use double quotes :D everything else seems fine, will give a new overall look tomorrow

ramon-bernardo commented:
> forgot to remove the `\'` now that we use double quotes :D everything else seems fine, will give a new overall look tomorrow

Removed! Thanks.

Why is the [assassin addon in portuguese](https://github.com/otland/forgottenserver/blob/48167ef5fee895f8195eebd82d15e6d01cece159/data/scripts/actions/others/dolls.lua#L58-L60)? _How strange, is that part of the story?_

EPuncker commented on data/scripts/eventcallbacks/monster/default_onDropLoot.lua:
```diff
@@ -12,7 +12,7 @@ ec.onDropLoot = function(self, corpse)
 			local item = corpse:createLootItem(monsterLoot[i])
 			if not item then
-				print('[Warning] DropLoot:', 'Could not add loot item to corpse.')
+				print("[Warning] DropLoot:", "Could not add loot item to corpse.")
```
```suggestion
				print("[Warning] DropLoot: Could not add loot item to corpse.")
```

EPuncker commented on data/scripts/actions/others/spellbook.lua:
```diff
@@ -18,7 +18,7 @@ function spellbook.onUse(player, item, fromPosition, target, toPosition, isHotke
 		if prevLevel ~= spell.level then
 			if i == 1 then
-				text[#text == nil and 1 or #text+1] = "Spells for Level "
+				text[not #text and 1 or #text+1] = "Spells for Level "
```
```suggestion
				text[not #text and 1 or #text + 1] = "Spells for Level "
```

EPuncker commented on data/scripts/actions/others/spellbook.lua:
```diff
@@ -18,7 +18,7 @@ function spellbook.onUse(player, item, fromPosition, target, toPosition, isHotke
-				text[#text == nil and 1 or #text+1] = "Spells for Level "
+				text[not #text and 1 or #text+1] = "Spells for Level "
 			else
 				text[#text+1] = "\nSpells for Level "
```
```suggestion
				text[#text + 1] = "\nSpells for Level "
```

ramon-bernardo commented:
My suggestions;

- Change total time `84600` to `24 * 60 * 60` and `3600` to `60 * 60`.
- `CONDITION_PARAM_TICKS, 150000` to `30 + (2 * 60 * 1000)`

EPuncker commented on data/npc/lib/npcsystem/modules.lua:
```diff
@@ -368,15 +368,15 @@ if not Modules then
 			end
 
 			if name and x and y and z and cost then
 				self:addDestination(name, {x=x, y=y, z=z}, cost, premium)
```
```suggestion
				self:addDestination(name, {x = x, y = y, z = z}, cost, premium)
```

EPuncker reviewed (changes requested):
always spaces between operators `z=z` to `z = z`

ramon-bernardo commented:
I finished my review, everything seems to be correct.

EPuncker commented:
can you resolve the conflict?

EPuncker commented:
okay one last thing I think we could have:

`.itemid` to `:getId()`
`.actionid` to `:getActionId()`
`.uid` to `:getUniqueId()`

ramon-bernardo commented:
I renamed some creaturescript events. i did an review on files from data, @EPuncker need another review.

EPuncker commented on data/actions/scripts/other/create_bread.lua:
```diff
@@ -4,13 +4,13 @@ local millstones = {1381, 1382, 1383, 1384}
 	local itemId = item:getId()
 	if itemId == 2692 then
-		if target.type == 1 and table.contains(liquidContainers, target.itemid) then
+		if target.type == 1 and table.contains(liquidContainers, target:getId()) then
```
sorry, I completely forgot about `.type` to `getSubType()`

ramon-bernardo replied:
target:getSubType() does not work, because getSubType() is for items.
I need to do a review on getUniqueId(). Wait.

EPuncker replied:
I think somehow, .type sometimes is used as getCount?

EPuncker reviewed (changes requested):
sorry, I completely forgot about .type to getSubType()

ramon-bernardo commented:
> sorry, I completely forgot about .type to getSubType()
> 
> also, according to our compat.lua ([here](https://github.com/otland/forgottenserver/blob/master/data/lib/compat/compat.lua#L43-L44)) the equivalent of is `player.uid``player:getId()`
> 
> also itemid and actionid values are different for creatures([npc, player, monster](https://github.com/otland/forgottenserver/blob/master/data/lib/compat/compat.lua#L55-L65)) and for items([container, teleport, item](https://github.com/otland/forgottenserver/blob/master/data/lib/compat/compat.lua#L68-L84))
> 
> (I'm learning a lot with this PR 😛 )

Yes! I realized this now while reviewing. I'm already fixing

EPuncker commented on data/actions/scripts/other/change_gold.lua:
```diff
@@ -6,7 +6,7 @@ local config = {
 function onUse(player, item, fromPosition, target, toPosition, isHotkey)
 	local coin = config[item:getId()]
-	if coin.changeTo and item.type == 100 then
+	if coin.changeTo and item:getSubType() == 100 then
```
shouldn't here be getCount instead? 🤔

ramon-bernardo replied:
I followed how [combat](https://github.com/otland/forgottenserver/blob/master/data/lib/compat/compat.lua#L76-L77) works, I also tested the script and it works normally.

ramon-bernardo replied:
But I will change it to `getCount()` . it is explicit

EPuncker commented on data/actions/scripts/other/create_bread.lua:
```diff
@@ -4,7 +4,7 @@ local millstones = {1381, 1382, 1383, 1384}
 	local itemId = item:getId()
 	if itemId == 2692 then
-		if target.type == 1 and table.contains(liquidContainers, target:getId()) then
+		if target:getSubType() == 1 and table.contains(liquidContainers, target:getId()) then
```
I think somehow, .type is sometimes used as getCount?

ramon-bernardo replied:
target:getSubType() does not work, correct is getCount()

EPuncker commented on data/actions/scripts/other/create_bread.lua:
```diff
@@ -4,7 +4,7 @@ local millstones = {1381, 1382, 1383, 1384}
 	local itemId = item:getId()
 	if itemId == 2692 then
-		if target:getSubType() == 1 and table.contains(liquidContainers, target:getId()) then
+		if target:getCount() == 1 and table.contains(liquidContainers, target:getId()) then
```
here I think it is subType for checking for water

ramon-bernardo replied:
getSubType()/getCount() does not work.
![image](https://user-images.githubusercontent.com/53269692/116491869-43f78d80-a871-11eb-906f-03591b741040.png)
![image](https://user-images.githubusercontent.com/53269692/116492110-d304a580-a871-11eb-86c2-25987f293a67.png)
Only `target.type` work.

Today I am going to review the scripts that I changed since the suggestion. It became a mess 👺

EPuncker replied:
yeah, I think we should keep the .type like it was before

EPuncker reviewed (changes requested):
and in the data/actions/scripts/other/fluids.lua entire file it is subtype because it is checking for liquids and not count, we need to test the fluids.lua ingame of the all files :P

EPuncker commented on data/actions/scripts/other/fluids.lua:
```diff
@@ -26,58 +26,58 @@ local fluidMessage = {
 			item:transform(item:getId(), 0)
 			return true
-		elseif target.type ~= 0 and item.type == 0 then
+		elseif target.type  ~= 0 and item:getCount() == 0 then
```
```suggestion
		elseif target.type ~= 0 and item:getCount() == 0 then
```

EPuncker commented on data/actions/scripts/other/fluids.lua:
```diff
@@ -26,58 +26,58 @@ local fluidMessage = {
 				player:sendCancelMessage("You have to process the bunch into the distillery to get rum.")
 			end
-		elseif item.type == 0 then
+		elseif item:getCount() == 0 then
```
here it is checking for fluid type, not count

ramon-bernardo replied:
Correct. I tested it with an empty vial.

EPuncker commented on data/actions/scripts/other/fluids.lua:
```diff
@@ -26,58 +26,58 @@ local fluidMessage = {
 				toPosition = player:getPosition()
 			end
-			Game.createItem(2016, item.type, toPosition):decay()
+			Game.createItem(2016, item:getCount(), toPosition):decay()
```
here I think it is creating with subtype as well

ramon-bernardo replied:
Correct. Item id 2016 is pool, I must use type and not quantity.

EPuncker commented on data/scripts/actions/others/glooth_bag.lua:
```diff
@@ -18,8 +18,8 @@ function gloothBag.onUse(player, item, fromPosition, target, toPosition, isHotke
 		if chance >= randomItem.chanceFrom and chance <= randomItem.chanceTo then
-			if randomItem.itemId then
-				local gift = randomItem.itemId
+			if randomItem.itemid then
```
actually the variable name here is itemId as you can check the config table

EPuncker commented on data/scripts/actions/others/gnomish_supply_package.lua:
```diff
@@ -33,8 +33,8 @@ function gnomish.onUse(player, item, fromPosition, target, toPosition, isHotkey)
 	for i = 1, #config do
 		local randomItem = config[i]
 		if chance >= randomItem.chanceFrom and chance <= randomItem.chanceTo then
-			if randomItem.itemId then
```
actually the variable name here is itemId as you can check the config table

EPuncker commented on data/scripts/actions/others/gooey_mass.lua:
```diff
@@ -17,8 +17,8 @@ function gooeyMass.onUse(player, item, fromPosition, target, toPosition, isHotke
 		if chance >= randomItem.chanceFrom and chance <= randomItem.chanceTo then
-			if randomItem.itemId then
-				local gift = randomItem.itemId
+			if randomItem.itemid then
```
actually the variable name here is itemId as you can check the config table

EPuncker commented on data/scripts/actions/others/muck_remover.lua:
```diff
@@ -22,10 +22,11 @@ function muckRemover.onUse(player, item, fromPosition, target, toPosition, isHot
 		randomItem = config[i]
 		if chance >= randomItem.from and chance <= randomItem.to then
 			if toPosition.x == CONTAINER_POSITION then
-				player:addItem(randomItem.itemId, randomItem.count or 1)
```
actually the variable name here is itemId as you can check the config table

EPuncker commented on data/scripts/actions/others/mysterious_remains.lua:
```diff
@@ -31,8 +31,8 @@ function mysteriousRemains.onUse(player, item, fromPosition, target, toPosition,
 		if chance >= randomItem.chanceFrom and chance <= randomItem.chanceTo then
-			if randomItem.itemId then
-				local gift = randomItem.itemId
+			if randomItem.itemid then
```
actually the variable name here is itemId as you can check the config table

EPuncker commented on data/scripts/actions/others/red_surprise_bag.lua:
```diff
@@ -24,8 +24,8 @@ function redSurpriseBag.onUse(player, item, fromPosition, target, toPosition, is
 		if chance >= randomItem.chanceFrom and chance <= randomItem.chanceTo then
-			if randomItem.itemId then
-				local gift = randomItem.itemId
+			if randomItem.itemid then
```
actually the variable name here is itemId as you can check the config table

EPuncker commented on data/scripts/actions/others/shaggy_ogre_bag.lua:
```diff
@@ -27,8 +27,8 @@ function shaggyBag.onUse(player, item, fromPosition, target, toPosition, isHotke
 		if chance >= randomItem.chanceFrom and chance <= randomItem.chanceTo then
-			if randomItem.itemId then
-				local gift = randomItem.itemId
+			if randomItem.itemid then
```
actually the variable name here is itemId as you can check the config table

EPuncker commented on data/scripts/actions/others/surprise_cube.lua:
```diff
@@ -29,8 +29,8 @@ function surpriseCube.onUse(player, item, fromPosition, target, toPosition, isHo
 		if chance >= randomItem.chanceFrom and chance <= randomItem.chanceTo then
-			if randomItem.itemId then
-				local gift = randomItem.itemId
+			if randomItem.itemid then
```
actually the variable name here is itemId as you can check the config table

EPuncker commented on data/scripts/actions/others/surprise_nest.lua:
```diff
@@ -19,8 +19,8 @@ function surpriseNest.onUse(player, item, fromPosition, target, toPosition, isHo
 	for i = 1, #config do
 		local randomItem = config[i]
 		if chance >= randomItem.chanceFrom and chance <= randomItem.chanceTo then
-			if randomItem.itemId then
```
actually the variable name here is itemId as you can check the config table

ramon-bernardo commented:
@EPuncker now, item ids follow the your name on the side. I noticed that this appeared in most of the files, i added missing ones.

EPuncker commented on data/movements/scripts/dough.lua:
```diff
@@ -1,9 +1,9 @@
 function onAddItem(moveitem, tileitem, position)
-	if moveitem:getId() == 2693 then
-		moveitem:transform(2689)
+	if moveitem:getId() == 2693 then -- timber wall
```
```suggestion
	if moveitem:getId() == 2693 then -- lump of dough
```

EPuncker commented on data/scripts/actions/others/clay_lump.lua:
```diff
@@ -1,7 +1,7 @@
-	{chance = {9.16, 25.48}, transformId = 11340, description = "It was made by |PLAYERNAME|, whose potter skills could use some serious improvement."},
+	{chance = {0.0, 1.54}, transformId = 11342, description = "This little figurine of Brog, the raging Titan, was skillfully made by |PLAYERNAME|.", achievement = true}, -- pretty clay statue
+	{chance = {1.54, 9.16}, transformId = 11341, description = "It was made by |PLAYERNAME| and is clearly a little figurine of.. hm, one does not recognise that yet."}, -- clay statue
+	{chance = {9.16, 25.48}, transformId = 11340, description = "It was made by |PLAYERNAME|, whose potter skills could use some serious improvement."}, --rough clay statue
```
```suggestion
	{chance = {9.16, 25.48}, transformId = 11340, description = "It was made by |PLAYERNAME|, whose potter skills could use some serious improvement."}, -- rough clay statue
```

EPuncker commented on data/scripts/actions/others/spider_egg.lua:
```diff
@@ -12,10 +12,10 @@ function spiderEgg.onUse(player, item, fromPosition, target, toPosition, isHotke
 end
 
-spiderEgg:id(7537)
+spiderEgg:id(7537) --spider egg
```
```suggestion
spiderEgg:id(7537) -- spider egg
```

EPuncker reviewed (changes requested):
still my only concern is the fluids script getCount() to .type

EPuncker commented on data/scripts/actions/tools/juice_squeezer.lua:
```diff
@@ -1,29 +1,14 @@
-	8840, -- raspberry
-	8841 -- lemon
-}
+local fruits = {2673, 2674, 2675, 2676, 2677, 2678, 2679, 2680, 2681, 2682, 5097, 8839, 8840, 8841}
```
I prefer the former with the names

EPuncker commented on data/spells/scripts/support/food.lua:
```diff
@@ -1,12 +1,4 @@
-	2690, -- roll
-	2696  -- cheese
-}
+local foods = {2666, 2671, 2681, 2674, 2689, 2690, 2696}
```
I prefer the former with the names

EPuncker reviewed (changes requested):
I prefer the former with the names

ramon-bernardo commented:
> I prefer the former with the names

I added the names next to ids; `waterIds = {}`  `holeId = {}` seem repeated

EPuncker commented on data/actions/scripts/tools/fishing.lua:
```diff
@@ -1,8 +1,60 @@
+}
+
+local lootCommon = {
+	2152, -- platinum coic
```
```suggestion
	2152, -- platinum coin
```

ramon-bernardo commented on data/scripts/actions/others/spellbook.lua:
```diff
@@ -32,5 +49,5 @@ function spellbook.onUse(player, item, fromPosition, target, toPosition, isHotke
 end
 
-spellbook:id(2175, 6120, 8900, 8901, 8902, 8903, 8904, 8918, 16112, 18401, 22422, 22423, 22424, 23771)
+spellbook:id(spellbooks)
```
It does not work. It is necessary to use `unpack()`.
